src/main/java/com/mongodb/hibernate/translate/attachment/Attachment.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.translate.attachment;
+
+import static java.lang.String.format;
+
+import com.mongodb.hibernate.internal.VisibleForTesting;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * Manages global state while translating SQL AST via {@link org.hibernate.sql.ast.SqlAstWalker}.
```

Comment:
"Manages global state" - I don't think that is correct. This utility provides a way to "return" values from `void` methods. It manages only its own object-local state, just like any other stateful object, and this does not require any special mentions.

Reply:
Not sure I agree with this, given the existence of fields like `AttachmentKey#COLUMN_NAME`.  We could make it true though by moving those static `AttachmentKey` instances into a separate class that is more closely tied to the AST walker.